usb/usb_host_uvc/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+configure_file(${CMAKE_CURRENT_LIST_DIR}/libuvc/include/libuvc/libuvc_config.h.in
+               ${CMAKE_CURRENT_BINARY_DIR}/include/libuvc/libuvc_config.h
+               @ONLY)
+
+set(LIBUVC_SOURCES libuvc/src/ctrl.c
+                   libuvc/src/ctrl-gen.c
+                   libuvc/src/device.c
+                   libuvc/src/diag.c
+                   libuvc/src/frame.c
+                   libuvc/src/init.c
+                   libuvc/src/misc.c
+                   libuvc/src/stream.c)
+
+idf_component_register(
+    SRCS ${LIBUVC_SOURCES} src/descriptor.c src/libusb_adapter.c
```

Comment:
Do you plan to separate `libusb_adapter` to a separate component in the future?

Reply:
Not necessarily, I would leave it as submodule. Do you see any reason why why we should?

Reply:
https://gitlab.espressif.cn:6688/espressif/esp-idf/-/merge_requests/14166#note_794407

Reply:
As @MaValki has explained to me later, fully matching libusb semantics would be pretty difficult. So we might just as well keep it as part of usb_host_uvc, until someone expresses interest in having a more generic libusb API supported.

Reply:
Thanks!